cmd/appgw-ingress/main.go

```diff
@@ -138,24 +138,27 @@ func main() {
 		azClient.SetAuthorizer(authorizer)
 	}
 
-	if _, err = azClient.GetGateway(); err != nil {
-		if controllererrors.IsErrorCode(err, controllererrors.ErrorApplicationGatewayNotFound) &&
-			env.EnableDeployAppGateway {
+	// check if application gateway exists/have get access
+	// When an AAD SP doesn't have access to the app gateway resource group, then AGIC can't read it's role assignments to look for the needed permission.
```

Comment:
nit: When an AAD SP doesn't have read access to the app gateway resource group...